vitrina/orgs/services.py

```diff
@@ -1,16 +1,150 @@
-from vitrina.orgs.models import Representative
+from enum import Enum
+from typing import Type
+
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import Model
+
+from vitrina.datasets.models import Dataset
+from vitrina.orgs.models import Representative, Organization
+from vitrina.requests.models import Request
+from vitrina.resources.models import DatasetDistribution
+from vitrina.users.models import User
 
 
 def has_coordinator_permission(user, organization):
     if user.is_authenticated:
         if user.is_staff:
             return True
-        if user.representative_set.filter(organization=organization).exists():
+        content_type = ContentType.objects.get_for_model(Organization)
+        if user.representative_set.filter(object_id=organization.pk, content_type=content_type).exists():
             representative = (
                 user.
                 representative_set.
-                filter(organization=organization).
+                filter(object_id=organization.pk, content_type=content_type).
                 first()
             )
             return representative.role == Representative.COORDINATOR
     return False
+
+
+class Action(Enum):
+    CREATE = 'create'
+    UPDATE = 'update'
+    DELETE = 'delete'
+
+
+def representative_exists(
+        user: User,
+        content_type: ContentType,
+        object_id: int,
+        role: str = None
```

Comment:
It would be better to use `Enum` for these kinds of things, because it is easy to lost the meaning and supported values for categorical data.

https://docs.djangoproject.com/en/3.2/ref/models/fields/#enumeration-types

```suggestion
        role: list[Representative.Role] | None = None
```

Also, it should be possible to check multiple roles, hence `list[]`.